requirements/customer/cli.tex

```diff
@@ -0,0 +1,102 @@
+
+\textsc{\large{\textbf{#1}}}
+\begin{itemize}
+\item[\textbf{Options:}] #2
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\newcommand{\jaoption}[3]{
+\begin{itemize}
+\item[] \textbf{#1} (#2)
+\item[\textbf{Effect:}] #3
+\end{itemize}
+}
+\subsection{Command Line Interface}
+All components of JobAdder are accessible through a command line interface (CLI).
+The command line interface on a machine with JobAdder installed has the following structure:
+\begin{equation}
+\text{<component name> <command> <options> <targets>},
+\end{equation}
+where the \textit{component name} can be any one of the following:
+\begin{itemize}
+\item \textbf{jobadder} to access the \textit{user client}.
+\item \textbf{jobcenter} to access the \textit{server}.
+\item \textbf{jobworker} to access the \textit{worker client}.
+\end{itemize}
+Commands, options, and valid targets are and options are specified in the following sections.
+
+Options can be added in the following ways:
+\begin{itemize}
+\item By supplying their names like "--<option\_x> --<option\_y>=2 --<option\_z>".
+\item By supplying their abbreviations like "-x -y 2 -z".
+\item By supplying their abbreviations like "-xz -y 2".
+\end{itemize}
+
+The available \textit{commands} are generally specific to the used component.
+Likewise, the available \textit{options} and \textit{targets} are generally specific to the commands of a used component.
+However, some commands and options are shared between components or commands.
+
+The JobAdder CLI ignores capitalization.
+JobAdder CLI options override the values of configuration files.
+If no command is given, the general help text for the given component is printed out.
+\subsubsection{Shared Options}
+\jaoption{config}{c}{Overrides the global configuration file with the given configuration file.}
+\jaoption{detach}{d}{Detaches the execution of the given command from the command line.}
+\jaoption{help}{h}{
+Prints out a list of the available commands. 
+When used in conjunction with a command:
+prints out more detailed information on the given command and a list of the available options.}
+\jaoption{verbose}{v}{
+Determines the amount of information printed out.
+If set to 2, detailed information is printed out.
+If set to 1, only high-level information is printed out (default).
+If set to 0, no information is printed out.}
+\subsubsection{User Client}
+\jacommand{add}
+{gpu, mount, name, priority, server, ram, threads}
+{Sends one or more job requests to the server.
+If targets are Docker containers the worker client simply runs the container.
+If targets are configuration files with a specified terminal command the worker executes the given terminal command.
+Otherwise targets are interpreted as terminal commands}
+\jaoption{gpu}{g}{
+If set to true, informs the server that the job requires a gpu.
+If set to a string that can be interpreted as VRAM amount, informs the server that a GPU with this amount of VRAM is required for the job.
```

Comment:
Need so specify `can be interpreted as a VRAM amount`, maybe simply `<N>G` and `<N>M` as formats?

Reply:
I didn't specify the amount because as far as I know the strings for storage/RAM representation are standardized (2G, 1024M, etc.).
I figured that there are already existing libraries for this conversion so I didn't specify further.

Reply:
Yeah, using a library makes sense. Maybe we can check available libraries and directly say here that we support the same formats as said library?